Makefile

```diff
@@ -1,38 +1,154 @@
-
-all: build 
-build: 
-	$(GOBUILD) -o $(CPE2CVE) ./cmd/$(CPE2CVE)/cpe2cve.go
-	$(GOBUILD) -o $(CSV2CPE) ./cmd/$(CSV2CPE)/csv2cpe.go
-	$(GOBUILD) -o $(NVDSYNC) ./cmd/$(NVDSYNC)/main.go
-	$(GOBUILD) -o $(RPM2CPE) ./cmd/$(RPM2CPE)/rpm2cpe.go
-
-clean: 
-	$(GOCLEAN)
-	rm -f $(CPE2CVE)
-	rm -f $(CSV2CPE)
-	rm -f $(NVDSYNC)
-	rm -f $(RPM2CPE)
-    
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+NAME = nvdtools
+VERSION = tip
+
+TOOLS = \
+	cpe2cve \
+	csv2cpe \
+	fireeye2nvd \
+	flexera2nvd \
+	nvdsync \
+	rpm2cpe \
+	rustsec2nvd \
+	vulndb
+
+DOCS = \
+	CODE_OF_CONDUCT.md \
+	CONTRIBUTING.md \
+	HOWTO.md \
+	LICENSE \
+	README.md
+
+GO = go
+GOOS = $(shell $(GO) env GOOS)
+GOARCH = $(shell $(GO) env GOARCH)
+
+TAR = tar
+ZIP = zip
+INSTALL = install
+
+# Compile all tools.
+all: $(TOOLS)
+
+# Compile TOOLS to ./build/bin/$tool using GOOS and GOARCH.
+$(TOOLS):
+	GOOS=$(GOOS) GOARCH=$(GOARCH) $(GO) build $(GOFLAGS) -o ./build/bin/$@ ./cmd/$@
+
+# Check/fetch all dependencies.
+deps:
+	GOOS=$(GOOS) GOARCH=$(GOARCH) $(GO) get -v -d ./...
+
+# install installs tools and documentation.
+# The install target is used by rpm and deb builders.
 install:
-	install -d $(DESTDIR)/usr/bin
-	install -p -m 0755 ./cpe2cve $(DESTDIR)/usr/bin/cpe2cve
-	install -p -m 0755 ./csv2cpe $(DESTDIR)/usr/bin/csv2cpe
-	install -p -m 0755 ./nvdsync $(DESTDIR)/usr/bin/nvdsync
-	install -p -m 0755 ./rpm2cpe $(DESTDIR)/usr/bin/rpm2cpe	 
-
-archive:
-	touch $(TGZ)
-	tar czf $(TGZ) --exclude=$(TGZ) --transform s/$(NAME)/$(PKG)/ ../$(NAME)
+	# tools
+	$(INSTALL) -d $(DESTDIR)/usr/bin
+	for tool in $(TOOLS); do $(INSTALL) -p -m 0755 ./build/bin/$$tool $(DESTDIR)/usr/bin/$$tool; done
+	# docs
+	$(INSTALL) -d $(DESTDIR)/usr/share/doc/nvdtools
+	for doc in $(DOCS); do $(INSTALL) -p -m 0644 $$doc $(DESTDIR)/usr/share/doc/nvdtools/$$doc; done
+
+DIST_NAME = $(NAME)-$(VERSION)
+DIST_DIR = build/$(DIST_NAME)
+
+# binary_dist creates a local binary distribution in DIST_DIR.
+binary_dist: $(TOOLS)
+	[ -d $(DIST_DIR) ] || mkdir -p $(DIST_DIR)/doc
+	cp $(DOCS) $(DIST_DIR)/doc
+	mv build/bin $(DIST_DIR)/bin
+
+# binary_tar creates tarball of binary distribution.
+binary_tar: binary_dist
+	[ -d build/tgz ] || mkdir build/tgz
+	cd build && $(TAR) czf tgz/$(NAME)_$(GOOS)_$(GOARCH)-$(VERSION).tar.gz $(DIST_NAME)
+	rm -rf $(DIST_DIR)
+
+# binary_zip creates zip of binary distribution.
+binary_zip: binary_dist
+	[ -d build/zip ] || mkdir build/zip
```

Comment:
this test (as well as others blow) is unnecessary: mkdir of exisiting directory is a no-op.

Reply:
It actually fails:

```
vagrant@ubuntu-bionic:/vagrant/nvdtools$ mkdir foo
vagrant@ubuntu-bionic:/vagrant/nvdtools$ mkdir foo
mkdir: cannot create directory ‘foo’: File exists
vagrant@ubuntu-bionic:/vagrant/nvdtools$ echo $?
1
```

Reply:
mkdir -p, sorry if it wasn't clear from my previous comment

Reply:
```
-p, --parents
              no error if existing, make parent directories as needed
```
(from man page)